### DISCUSSION TITLE
API v1 vs v2 endpoints

### DISCUSSION THREAD

**Discussion by dreautall** [firefly-iii/firefly-iii#8340] [General]
Hi there 👋 

Quick question as I see more and more endpoints added to the API v2 that already exist in a "v1 version".

In general, is there any immediate benefit of switching to the new v2 calls vs. the old v1 calls? Like performance, future stability or similar? I just checked the (fairly simple) Autocomplete API call in the PHP code, and looks like despite a limit parameter they are the same, so it wouldn't really make sense to "switch" in my Apps code to the new call. Not sure about the other ones though (like for example performance improvements, or stuff like #8323 only beeing implemented in the new call).

Thank you!

**Comment by JC5** [comment]
Most of them are still in development and barely documented. I would not use them yet. 

In the future Firefly III will support multiple financial administrations, where you can share your administration with another Firefly III user. The current endpoints (v1) will not support this, but the new endpoints (v2) are all based on this new database structure.

Once this goes live, you should be able to switch from v1 to v2 fairly seamlessly.

**Reply by dreautall** [reply]
Alright thanks for the feedback, I'll hold off then from switching to them for now unless I really need a new API feature for something. Thank you!
